Library/Homebrew/utils/bottles.rb

```diff
@@ -0,0 +1,106 @@
+require "tab"
+require "extend/ARGV"
+
+module Utils
+  module Bottles
+    class << self
+      def tag
+        @bottle_tag ||= RUBY_PLATFORM.to_sym
+      end
+
+      def built_as?(f)
+        return false unless f.installed?
+        tab = Tab.for_keg(f.installed_prefix)
+        tab.built_as_bottle
+      end
+
+      def file_outdated?(f, file)
+        filename = file.basename.to_s
+        return unless f.bottle && filename.match(Pathname::BOTTLE_EXTNAME_RX)
+
+        bottle_ext = filename[bottle_native_regex, 1]
+        bottle_url_ext = f.bottle.url[bottle_native_regex, 1]
+
+        bottle_ext && bottle_url_ext && bottle_ext != bottle_url_ext
+      end
+
+      def native_regex
+        /(\.#{tag}\.bottle\.(\d+\.)?tar\.gz)$/o
+      end
+
+      def receipt_path(bottle_file)
+        Utils.popen_read("/usr/bin/tar", "-tzf", bottle_file, "*/*/INSTALL_RECEIPT.json").chomp
```

Comment:
I prefer to use `tar` and other utilities from `PATH`. Or alternatively, prefer the brewed utility over the host's utility, found in either `/usr/bin/` or `/bin`. Either `PATH` could be modified to search these three directories, or we could add our own path searching wrapper function.